packages/integration-tests/src/tests/api/hook/hook.data.hook.trigger.test.ts

```diff
@@ -0,0 +1,341 @@
+import {
+  RoleType,
+  hookEventGuard,
+  hookEvents,
+  jsonGuard,
+  managementApiHooksRegistration,
+  type Hook,
+  type Role,
+} from '@logto/schemas';
+import { z } from 'zod';
+
+import { authedAdminApi } from '#src/api/api.js';
+import { createResource } from '#src/api/resource.js';
+import { createScope } from '#src/api/scope.js';
+import {
+  OrganizationApiTest,
+  OrganizationRoleApiTest,
+  OrganizationScopeApiTest,
+} from '#src/helpers/organization.js';
+import { UserApiTest, generateNewUser } from '#src/helpers/user.js';
+import { generateName, waitFor } from '#src/utils.js';
+
+import WebhookMockServer from './WebhookMockServer.js';
+import {
+  organizationDataHookTestCases,
+  organizationRoleDataHookTestCases,
+  organizationScopeDataHookTestCases,
+  roleDataHookTestCases,
+  scopesDataHookTestCases,
+  userDataHookTestCases,
+} from './test-cases.js';
+
+const mockHookResponseGuard = z.object({
+  signature: z.string(),
+  payload: z.object({
+    event: hookEventGuard,
+    createdAt: z.string(),
+    hookId: z.string(),
+    body: jsonGuard.optional(),
+    method: z
+      .string()
+      .optional()
+      .transform((value) => value?.toUpperCase()),
+    _matchedRoute: z.string().optional(),
+  }),
+});
+
+type MockHookResponse = z.infer<typeof mockHookResponseGuard>;
+
+describe('Trigger DataHook through management API', () => {
+  const hookName = 'management-api-hook';
+  const webhooks = new Map<string, Hook>();
+  const webhookResults = new Map<string, MockHookResponse>();
+
+  // Record the hook response to the webhookResults map.
+  // Compare the webhookResults map with the managementApiHooksRegistration to verify all hook is triggered.
+  const webhookResponseHandler = (response: string) => {
+    const result = mockHookResponseGuard.parse(JSON.parse(response));
+    const { payload } = result;
+
+    // Use _matchedRoute as the key
+    if (payload._matchedRoute) {
+      webhookResults.set(`${payload.method} ${payload._matchedRoute}`, result);
+    }
+  };
+
+  const getWebhookResult = async (key: string) => {
+    // Wait for 100ms to ensure all webhook responses are received.
+    await waitFor(100);
```

Comment:
can be shorter like 10 or 20

Reply:
10 is too short for our ci machine. set to 10 the test will fail.